LoopKit/Onboarding.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  Onboarding.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 12/10/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+public protocol Onboarding: AnyObject {
+    /// The unique identifier of this type of onboarding.
+    var onboardingIdentifier: String { get }
+
+    /// Initializes the onboarding.
+    init()
```

Comment:
Yes, protocol conformation would require a `init()` implementation. It does not initiate the onboarding (just creates it). I will clarify in the comment.

Reply:
I can migrate to a static factory method (rather that `init()`).

However I cannot get rid of the object entirely as we need an object that keeps track of onboarding state and can report back to Loop (so it can persist it). This falls into the "I already know I'm going to need it in my next work so rather than do the work twice I'll just do it here" category.

Basically, we need an object much like a service, CGM manager, and pump manager, that can be instantiated both brand-new and a previously persisted state so we can "resume" onboarding.